src/app/cakes/handlers/types.ts

```diff
@@ -0,0 +1,18 @@
+import { Request, Response } from 'express';
+import { ICake } from '../cake.class';
+
+type Params = {
+    id?: string
+};
+type Query = {
+};
+
+type Req = Request<Params, {}, ICake, Query>;
+type Res = Response;
+
+export {
+    Params,
+    Query,
+    Req,
+    Res
+}
```

Comment:
I think these types should be created in a separate file since cannot be shared.

You will have multiple request handlers for multiples verbs, you cannot share the query params from a `get` in the `post`.